api/client/contextdialer.go

```diff
@@ -364,6 +364,9 @@ func newTLSRoutingWithConnUpgradeDialer(ssh ssh.ClientConfig, params connectPara
 			},
 			ALPNConnUpgradeRequired: IsALPNConnUpgradeRequired(ctx, params.addr, insecure),
 			GetClusterCAs: func(_ context.Context) (*x509.CertPool, error) {
+				// TODO(noah): there's potentially a bug here when multiple
+				// credentials have been configured as the CAs are always
+				// retrieved from the first credential.
```

Comment:
hmm.. good catch. Probably can just use `params.tlsConfig.RootCAs`?

Reply:
That looks like a good fix - I'll pop that in and test it :)

Reply:
Works nicely 🥳  [b22c6e5](https://github.com/gravitational/teleport/pull/36411/commits/b22c6e585fc857ff0a9f56dcf74af2dbfed71100)

Reply:
thanks! 🙏